flink-python/pyflink/table/expressions.py

```diff
@@ -544,6 +544,25 @@ def if_then_else(condition: Union[bool, Expression[bool]], if_true, if_false) ->
     return _ternary_op("ifThenElse", condition, if_true, if_false)
 
 
+def coalesce(*args) -> Expression:
+    """
+    Coalesce specifies a series of expressions, and returns the first expression whose value is
+    not null. If all the expressions evaluate as null, coalesce returns a null value. The return
+    type of the coalesce function is the least common type among its arguments, and it's nullable
+    if all its arguments are nullable.
+
+    e.g. coalesce(null, "-") returns "-".
```

Comment:
Here, too, please refer to e.g. the `json_*` functions on how we want to format the example sections. Same for Java & Scala.